packages/medusa/src/types/payment-processor.ts

```diff
@@ -0,0 +1,26 @@
+import { Address, Customer, ShippingMethod } from "../models"
+
+export type PaymentProcessorContext = {
+  cart?: {
+    context: Record<string, unknown>
+    id: string
+    email: string
+    shipping_address: Address | null
+    shipping_methods: ShippingMethod[]
+  }
+  currency_code: string
+  amount: number
+  resource_id?: string
+  customer?: Customer
```

Comment:
question: Do you think we should allow for `email` and `[shipping | billing]_address` without a cart? Potentially also `context`

Reply:
ahah, partly part of my previous comment